test/NuGet.Core.FuncTests/NuGet.Commands.FuncTest/RestoreCommandTests.cs

```diff
@@ -4089,20 +4090,19 @@ public async Task Restore_WithHttpSource_Warns(string allowInsecureConnections,
             var result = await command.ExecuteAsync();
 
             // Assert
-            result.Success.Should().BeTrue(because: logger.ShowMessages());
-            result.LockFile.Libraries.Should().HaveCount(0);
-
-            string expectedWarning = $"You are running the 'restore' operation with an 'HTTP' source, '{httpSourceUrl}'. Non-HTTPS access will be removed in a future version. Consider migrating to an 'HTTPS' source.";
+            string expectedError = string.Format(CultureInfo.CurrentCulture, NuGet.PackageManagement.Strings.Error_HttpSource_Single, "restore", httpSourceUrl);
 
-            if (isHttpWarningExpected)
+            if (isHttpErrorExpected)
             {
+                result.Success.Should().BeFalse(because: logger.ShowMessages());
                 result.LockFile.LogMessages.Should().HaveCount(1);
                 IAssetsLogMessage logMessage = result.LockFile.LogMessages[0];
-                logMessage.Code.Should().Be(NuGetLogCode.NU1803);
-                Assert.Equal(expectedWarning, logMessage.Message);
+                logMessage.Code.Should().Be(NuGetLogCode.NU1302);
+                Assert.Equal(expectedError, logMessage.Message);
             }
             else
             {
+                result.Success.Should().BeTrue(because: logger.ShowMessages());
```

Comment:
I know this change has not been introduced in this PR, but it is a bit confusing that the test name says `ThrowError` but there is a code path where test actually succeeds. May be in a follow-up PR.